roles/acm_attach_spoke_cluster/tasks/main.yaml

```diff
@@ -0,0 +1,121 @@
+---
+
+- name: Assert the required variables are defined
+  ansible.builtin.assert:
+    that:
+      - aasc_cluster_kubeconfig_path is defined
+      - aasc_cluster_kubeconfig_path | length > 0
+      - aasc_cluster_name is defined
+      - aasc_cluster_name | length > 0
+
+# This will automatically create the cluster namespace, called
+# aasc_cluster_name too.
+- name: Create ManagedCluster
+  community.kubernetes.k8s:
+    definition:
+      apiVersion: cluster.open-cluster-management.io/v1
+      kind: ManagedCluster
+      metadata:
+        name: "{{ aasc_cluster_name }}"
+        labels:
+          name: "{{ aasc_cluster_name }}"
+          cloud: auto-detect
+          vendor: auto-detect
+        annotations: {}
+      spec:
+        hubAcceptsClient: true
+
+- name: Ensure spoke cluster namespace has been created
+  community.kubernetes.k8s_info:
+    api: v1
+    kind: Namespace
+    name: "{{ aasc_cluster_name }}"
+  register: _aasc_namespace_status
+  until:
+    - _aasc_namespace_status.resources | length > 0
+  retries: 60
+  delay: 10
+
+# To import the spoke cluster to the hub cluster, we need to create
+# a secret that contains the spoke cluster's kubeconfig.
+# This requires a special formatting, including double line breaks
+# and a correct indentation.
+- name: Create autoimport secret
+  block:
+    - name: Create temporary template directory
+      ansible.builtin.tempfile:
+        state: directory
+        suffix: build
+      register: _aasc_temp_dir
+
+    - name: Get kubeconfig content
+      ansible.builtin.slurp:
```

Comment:
set_fact can be simpler:
`ansible.builtin.set_fact:
     _assc_kubeconfig_content: "{{file_contents: "{{ lookup('file', aasc_cluster_kubeconfig_path) }}"}}"`

Reply:
the lookup plugin will work only when it's localhost, slurp is more flexible and supports any host

Reply:
As @sshnaidm said, I'd prefer to use `slurp` because of its flexibility